src/pages/Vaults/VaultsOverview/components/CreateVaultWizard/DespositCollateralStep.tsx

```diff
@@ -0,0 +1,127 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { CollateralCurrencyExt, newMonetaryAmount } from '@interlay/interbtc-api';
+import { MonetaryAmount } from '@interlay/monetary-js';
+import { useId } from '@react-aria/utils';
+import Big from 'big.js';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { useMutation } from 'react-query';
+import * as z from 'zod';
+
+import { displayMonetaryAmountInUSDFormat, parseLocaleNumber } from '@/common/utils/utils';
+import { CTA, Span, Stack, TokenField } from '@/component-library';
+import ErrorModal from '@/components/ErrorModal';
+import { GOVERNANCE_TOKEN } from '@/config/relay-chains';
+import { getErrorMessage } from '@/utils/helpers/forms';
+import { validateDepositCollateral } from '@/utils/validation/vault/create';
+
+import { useDepositCollateral } from '../../utils/use-deposit-collateral';
+import { StyledDd, StyledDepositTitle, StyledDItem, StyledDl, StyledDt, StyledHr } from './CreateVaultWizard.styles';
+import { StepComponentProps, withStep } from './Step';
+
+const DEPOSIT_COLLATERAL_AMOUNT = 'deposit-collateral-amount';
+
+type CollateralFormData = { [DEPOSIT_COLLATERAL_AMOUNT]: string };
+
+type Props = {
+  collateralCurrency: CollateralCurrencyExt;
+  minCollateralAmount: Big;
+  onSuccessfulDeposit?: () => void;
+};
+
+type DespositCollateralStepProps = Props & StepComponentProps;
+
+const DepositCollateralStep = ({
+  onSuccessfulDeposit,
+  collateralCurrency,
+  minCollateralAmount
+}: DespositCollateralStepProps): JSX.Element => {
+  const titleId = useId();
+  const { t } = useTranslation();
+  const { collateral, fee, governance } = useDepositCollateral(collateralCurrency, minCollateralAmount);
+
+  const validationParams = {
+    minAmount: collateral.min.raw,
+    balance: collateral.balance.raw,
+    governanceBalance: governance.raw,
+    transactionFee: fee.raw
+  };
+  const schema = z.object({
+    [DEPOSIT_COLLATERAL_AMOUNT]: validateDepositCollateral(t, validationParams)
+  });
+  const {
+    register,
+    handleSubmit: h,
+    watch,
+    formState: { errors, isValid }
+  } = useForm<CollateralFormData>({
+    mode: 'onChange',
+    resolver: zodResolver(schema)
+  });
+
+  const registerNewVaultMutation = useMutation<void, Error, MonetaryAmount<CollateralCurrencyExt>>(
+    (collateralAmount) => window.bridge.vaults.registerNewCollateralVault(collateralAmount),
+    {
+      onSuccess: onSuccessfulDeposit
+    }
+  );
+
+  const inputCollateral = parseLocaleNumber(watch(DEPOSIT_COLLATERAL_AMOUNT) || '0');
```

Comment:
@tomjeatt, instead of this `parseLocaleNumber`, we could work on a feature for the NumberInput.
We should be careful handling these inputs.

Reply:
Yes - if you could take a look at the changes we made yesterday, our aim here should be to format numbers only at the point at which they're displayed (i.e. as late as possible!) so that we avoid using formatted/parsed numbers for transactions/validation.

Only exception is when we're rounding or truncating decimals.

I'll leave this to your judgement, but if you think there's any risk of errors or regressions, then this should be done ahead of merging.